validation/pipeline/qase/schemaupload/main.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/rancher/tests/validation/pipeline/qase"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	_, callerFilePath, _, _ = runtime.Caller(0)
+	basepath                = filepath.Join(filepath.Dir(callerFilePath), "..", "..", "..", "..")
+)
+
+func main() {
+	qase.SetupQaseClient()
+
+	schemaMap, err := getRawSchemaData()
+	if err != nil {
+		logrus.Error("Error retrieving test schemas: ", err)
+		return
+	}
+
+	for project, schemas := range schemaMap {
+		var cases []qase.TestCase
+		for _, schema := range schemas {
+			suite := extractSubstrings(schema, `## Test Suite: (.+)\n`)[0][1]
+			testSuite, suiteErr := qase.GetTestSuite(project, suite)
+			testSuiteId := testSuite.Id
+			if suiteErr != nil {
+				testSuiteId, _ = qase.CreateTestSuite(project, suite)
+			}
+			parsedCases, err := parseSchema(schema, testSuiteId)
+			if err != nil {
+				logrus.Error("Error parsing schemas: ", err)
+				return
+			}
+			cases = append(cases, parsedCases...)
+		}
+
+		err = qase.UploadTests(project, cases)
+		if err != nil {
+			logrus.Errorln("Error uploading tests:", err)
+		}
+	}
+}
+
+// getRawSchemaData retrieves the tests from schemas.md files defined within each Go package and returns a map of the string content within each suite
```

Comment:
might be helpful to say what the expected contents of the map are, i.e. 
[projectName] : []{testCaseNames}